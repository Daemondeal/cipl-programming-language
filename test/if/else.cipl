if true: println("good") 
else: println("bad")

if false: println("bad") 
else: println("good")

if false:
    pass
else:
    println("block")