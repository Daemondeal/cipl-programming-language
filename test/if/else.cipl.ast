(if true (then (expression (call print good)) (else (expression (call print bad)))
(if false (then (expression (call print bad)) (else (expression (call print good)))
(if false (then (block
    pass
) (else (block
    (expression (call print block))
))

