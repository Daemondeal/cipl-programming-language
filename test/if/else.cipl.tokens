Token          Lexeme         Line 
IF             if             1    
TRUE           true           1    
COLON          :              1    
IDENTIFIER     println        1    
LEFT_PAREN     (              1    
STRING         "good"         1     good
RIGHT_PAREN    )              1    
LINE_END       \n             2    
ELSE           else           2    
COLON          :              2    
IDENTIFIER     println        2    
LEFT_PAREN     (              2    
STRING         "bad"          2     bad
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IF             if             4    
FALSE          false          4    
COLON          :              4    
IDENTIFIER     println        4    
LEFT_PAREN     (              4    
STRING         "bad"          4     bad
RIGHT_PAREN    )              4    
LINE_END       \n             5    
ELSE           else           5    
COLON          :              5    
IDENTIFIER     println        5    
LEFT_PAREN     (              5    
STRING         "good"         5     good
RIGHT_PAREN    )              5    
LINE_END       \n             6    
IF             if             7    
FALSE          false          7    
COLON          :              7    
LINE_END       \n             8    
INDENT                        8    
PASS           pass           8    
LINE_END       \n             9    
DEDENT                        9    
ELSE           else           9    
COLON          :              9    
LINE_END       \n             10   
INDENT                        10   
IDENTIFIER     println        10   
LEFT_PAREN     (              10   
STRING         "block"        10    block
RIGHT_PAREN    )              10   
LINE_END       \n             10   
DEDENT                        10   
EOF                           10   
