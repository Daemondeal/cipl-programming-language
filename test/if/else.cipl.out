::stdout::
good
good
block

::stderr::
